File: Array/485. Max Consecutive Ones/Solution.java
public class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        if (nums == null || nums.length == 0) {
            return 0;
        }
        int count = nums[0] == 1 ? 1 : 0;
        int max = count;

        for (int i = 1; i < nums.length; i++) {
            if (nums[i] == 1) {
                count += 1;
            } else {
                count = 0;
            }
            max = Math.max(max, count);
        }
        return max;
    }
}

File: Array/485. Max Consecutive Ones/Solution.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMaxConsecutiveOnes = function(nums) {
    if (nums == 0 || nums.length == 0) {
        return 0
    }
    
    let result = 0;
    let count = 0;
    
    for(i = 0; i<nums.length; i++) {
        if(nums[i] === 0) {
            count = 0
        } else {
            count ++
        }
        result = Math.max(result, count)
    }
    return result
};